url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgPattern.java

```diff
@@ -0,0 +1,633 @@
+        int argCount = nonOptionList.size();
+        ArgNode node = null;
+        String arg = null;
+        failBase = null;
+        failNode = null;
+        int nodeIndex = 0;
+        for (int n = 0; n < pattern.length(); n++) {
+            char c = pattern.charAt(n);
+
+            // Literal.
+            if (c == 'L') {
+
+                // Get the literal text.
+                node = patternList.get(nodeIndex++);
+                ArgOperand operand = node.getOperand();
+                String literal = operand.getName();
+
+                // Next non-option argument.
+                if (argIndex >= argCount) {
+                    failNode = node;
+                    if (failBase == null) {
+                        failBase = failNode;
+                    }
+                    return false;
+                }
+                arg = nonOptionList.get(argIndex++);
+                trace("compare", literal + " to " + arg);
+
+                // Ignore case. Not a match.
+                if (!arg.equalsIgnoreCase(literal)) {
+                    failNode = node;
+                    if (failBase == null) {
+                        failBase = failNode;
+                    }
+                    return false;
+                }
+            }
+
+            // Variable.
+            else if (c == 'V') {
+            	
+            	// Get the variable node.
+            	node = patternList.get(nodeIndex++);
+
+                // Accept the next non-option argument.
+                if (argIndex >= argCount) {
+                    failNode = node;
+                    if (failBase == null) {
+                        failBase = failNode;
+                    }
+                    return false;
+                }
+                arg = nonOptionList.get(argIndex++);
+                trace("Variable", arg);
+            }
+            
+            // Base node.
+            else if (c == 'b') {
+                failBase = patternList.get(nodeIndex++);
+            }
+            
+            // Group node.
+            else if (c == 'g') {
+                ++nodeIndex;
+            }
+            
+            // Reduced optional argument.
+            else if (c == 'x') {
+                ++nodeIndex;
+            }
+        }
+        if (patternMin > argCount || argCount > patternMax) {
+            failBase = patternList.get(0);
+            failNode = failBase;
+            return false;
+        }
+        return true;
+    }
+    
+    private String buildPattern (ArgNode start, StringBuilder sb) {
+        argList.setCurrent(start);
+        for (ArgNode node = start; node != null; node = argList.goEast()) {
+            if (node.isOperand()) {
+                buildOperand(node, sb);
+            } else if (node.isGroup()) {
+                buildGroup(node, sb, false);
+            }
+        }
+        return sb.toString();
+    }
+
+    private String rebuildPattern (ArgNode start, StringBuilder sb) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.